api/v1alpha1/api.go

```diff
@@ -144,3 +166,87 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// LLMProviderAuthenticationType specifies the type of auth mechanism used to access a Provider.
+type LLMProviderAuthenticationType string
+
+const (
+	LLMProviderAuthenticationTypeAPIKey    LLMProviderAuthenticationType = "APIKey"
+	LLMProviderAuthenticationTypeOIDC      LLMProviderAuthenticationType = "OIDC"
+	LLMProviderAuthenticationTypeStaticKey LLMProviderAuthenticationType = "StaticKey"
+)
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicy specifies configuration for authentication and authorization rules on the traffic
+// exiting the gateway to the backend.
+type BackendSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              BackendSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// BackendSecurityPolicySpec specifies authentication and authorization rules on access the provider from the Gateway.
+type BackendSecurityPolicySpec struct {
+	// Type specifies the auth mechanism used to access the provider. Currently, only "APIKey", "StaticKey", AND "OIDC" are supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey;StaticKey;OIDC
+	Type LLMProviderAuthenticationType `json:"type"`
+
+	// APIKey specific configuration. The API key will be injected into the Authorization header.
+	// +optional
+	APIKey *LLMProviderAPIKey `json:"apiKey,omitempty"`
+
+	// OIDC tokens are retrieved from the following configuration. The token and backend type will determine how Authorization is configured.
+	// +optional
+	OIDC *egv1a1.OIDC `json:"oidc,omitempty"`
```

Comment:
API key and OIDC are not AI traffic specifc, is it possible to reuse the EG SecurityPolicy?

Reply:
Oh, I see, this is for LLM Backend authentication, and EG SecurityPolicy is used for client side authentication, right?

Reply:
yea